test/system/posts_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'application_system_test_case'
+
+class PostsTest < ApplicationSystemTestCase
+  setup do
+    @user = users(:one)
+    @post = posts(:one)
+  end
+
+  test 'posts index' do
+    visit user_posts_url(@user)
+
+    assert_selector 'img', class: 'user-image'
+    assert_selector 'h2', text: @user.name
+    assert_selector 'p', text: "Number of posts: #{@user.post_counter}"
+
+    # Test to see if all posts are displayed on the index page
+    @user.posts.order(created_at: :desc).each do |post|
+      assert_selector '.user-post-title', text: post.title
+      assert_selector '.user-post-text', text: post.text
+      assert_selector '.counter', text: "Comments: #{post.comments_counter || 0}, Likes: #{post.likes_counter || 0}"
+      assert_selector '.comments-card', text: post.recent_comments.reverse.first.text
+      click_on post.title
+
+      # Check that you are now on the show page for the post
+      assert_current_path user_post_path(@user, post)
+      click_on 'Back to Post index'
+    end
+  end
```

Comment:
- [ ] Kindly write the test for `I can see a section for pagination if there are more posts than fit on the view.`. That is missing from the tests at this moment.

Reply:
I actually had a doubt about this issue. What does the pagination do? Should it be a button that expands the page? Or should it be a functioning pagination with numbers and links? Because that was not listed in the requirements and there's really not much to go from the template:
![image](https://github.com/tomasesquivelgc/Blog-App/assets/117373513/c2a4605e-be67-4ef6-ab82-f8359584a09c)

I was told by previous reviewers that the posts_index should display all of the posts, so by adding the pagination should I also limit the number of posts that are displayed?
Some help to clear this doubts would be greatly appreciated.
Cheers.